test/unit/test-mysqlQuery.js

```diff
@@ -220,6 +310,17 @@ describe('MysqlQuery', function(){
             Should.exist(error);
             error.message.should.eql("Invalid DELETE query");
         });
+        it('Should throw an error for INSERT ... ON DUPLICATE KEY UPDATE queries', function() {
+            const query = "INSERT INTO users (first_name, last_name, password) VALUES ('firstName', 'lastName', 'boom!') ON DUPLICATE KEY UPDATE first_name = 'firstName', last_name = 'lastName', password = 'boom!'";
+            let error = null;
+            try {
+                const result = MysqlQuery.validateDelete(query);
+            } catch(e) {
+                error = e;
+            }
+            Should.exist(error);
+            error.message.should.eql("Invalid DELETE query");
+        });
```

Comment:
Minor, but this code looks like it is repeated a lot in this file. Next time someone is in here they could DRY it up.